src/NServiceBus.Core/Pipeline/Incoming/ReceivePipelineConfigExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    /// <summary>
+    /// Provides access to receive pipeline notifications.
+    /// </summary>
+    public static class ReceivePipelineConfigExtensions
+    {
+        /// <summary>
+        /// Subscribes to notifications for completed receive pipelines.
+        /// </summary>
+        public static void OnReceivePipelineCompleted(this PipelineSettings pipelineSettings, Func<ReceivePipelineCompleted, Task> subscription)
```

Comment:
not against that approach, but I dislike that we now have two different public event subscription APIs (well, three including the legacy API). So there are some events registered via `config.Pipeline` and some via `config.Notifications` ? 👎